src/grt/src/GlobalRouter.cpp

```diff
@@ -2614,24 +2590,24 @@ Net* GlobalRouter::getNet(odb::dbNet* db_net)
 
 void GlobalRouter::getNetsByType(NetType type, std::vector<Net*>& nets)
 {
-  if (type == NetType::Clock || type == NetType::Signal) {
-    bool get_clock = type == NetType::Clock;
-    for (Net net : *nets_) {
-      if ((get_clock && net.getSignalType() == odb::dbSigType::CLOCK
-           && !clockHasLeafITerm(net.getDbNet()))
-          || (!get_clock
-              && (net.getSignalType() != odb::dbSigType::CLOCK
-                  || clockHasLeafITerm(net.getDbNet())))) {
-        nets.push_back(db_net_map_[net.getDbNet()]);
-      }
-    }
-  } else if (type == NetType::Antenna) {
+  if (type == NetType::Antenna) {
     for (odb::dbNet* db_net : dirty_nets_) {
       nets.push_back(db_net_map_[db_net]);
     }
   } else {
+    // add clock nets not connected to a leaf first    
+    for (Net net : *nets_) {
+      bool has_leaf = clockHasLeafITerm(net.getDbNet());
+      if ((net.getSignalType() == odb::dbSigType::CLOCK && !has_leaf)) {
```

Comment:
Here and in initializeNets it looks like it would be more useful to have isNonLeafClock rather than clockHasLeafITerm and some extra logic.